src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java

```diff
@@ -65,26 +66,31 @@ public void run() {
      *
      * @param context the context to stop
      */
-    public static synchronized void stopSparkContext(JavaSparkContext context) {
+    public static synchronized void stopSparkContext(final JavaSparkContext context) {
         // only call stop for a non-test context
         if (context != testContext) {
             context.stop();
         }
     }
 
-    private static JavaSparkContext createSparkContext(String appName, String master) {
-        SparkConf sparkConf = new SparkConf().setAppName(appName)
+    private static JavaSparkContext createSparkContext(final String appName, final String master) {
+        final SparkConf sparkConf = new SparkConf().setAppName(appName)
                 .setMaster(master)
                 .set("spark.serializer", KryoSerializer.class.getCanonicalName())
                 .set("spark.kryo.registrator", "org.broadinstitute.hellbender.engine.spark.GATKRegistrator")
-                .set("spark.kryoserializer.buffer.max", "512m");
+                .set("spark.kryoserializer.buffer.max", "512m")
+                .set("spark.driver.maxResultSize", "0")
+                .set("spark.driver.userClassPathFirst", "true")
+                .set("spark.executor.userClassPathFirst", "true")
+                .set("spark.io.compression.codec", "lzf")
+                .set("spark.yarn.executor.memoryOverhead", "600");
```

Comment:
Create ticket to eventually expose these (either as args in `SparkCommandLineProgram`, or via the `spark-submit`/`gcloud` wrapper that you're writing).